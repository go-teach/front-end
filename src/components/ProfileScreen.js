import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { StyleSheet, Text, View } from 'react-native';

import LoginStatusMessage from './LoginStatusMessage';

const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: 'center',
		alignItems: 'center',
		backgroundColor: '#F5FCFF',
	},
	welcome: {
		fontSize: 20,
		textAlign: 'center',
		margin: 10,
	},
});

const ProfileScreen = ({ isLoggedIn }) => (
	<View style={styles.container}>
		<Text style={styles.welcome}> Profile Screen </Text>
		<LoginStatusMessage isLoggedIn={isLoggedIn} />
	</View>
);

ProfileScreen.propTypes = {
	isLoggedIn: PropTypes.bool.isRequired,
};

ProfileScreen.navigationOptions = {
	title: 'Profile',
};


const mapStateToProps = state => ({
	isLoggedIn: state.auth.isLoggedIn,
});

export { ProfileScreen }; //for testing
export default connect(mapStateToProps)(ProfileScreen);
