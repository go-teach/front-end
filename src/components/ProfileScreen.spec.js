import React from 'react';
import { shallow } from 'enzyme';
import { ProfileScreen } from './ProfileScreen';

describe('ProfileScreen component', () => {
	it('should return proper component', () => {
		const wrapper = shallow(<ProfileScreen isLoggedIn />);
		expect(wrapper).toMatchSnapshot();
	});
});
