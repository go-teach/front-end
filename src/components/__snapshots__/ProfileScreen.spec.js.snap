// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen component should return proper component 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
     Profile Screen 
  </Text>
  <LoginStatusMessage
    isLoggedIn={true}
  />
</Component>
`;
